### DISCUSSION TITLE
Are "forwarding" queries never cached?

### DISCUSSION THREAD

**Discussion by CampionLoong** [DNSCrypt/dnscrypt-proxy#1880] [Q&A]
I noticed in the log that queries to names in the "forwarding rules" file are almost never cached. Is this really the case? And if so, can results from those forwarded queries be cached like regular ones to reduce upstream traffic? This is mainly for performance

**Comment by jedisct1** [comment]
Forwarding was implemented to send queries for the local network to the router. In that context, caching doesn't really matter; pretty much the opposite: new devices should show up and disappear immediately.

**Reply by CampionLoong** [reply]
My use case is less with local network, but routing WAN names to specific resolvers (i.e. the 'example.com' case in forwarding_rules's comments).

Are those results cached? Shouldn't they be?

**Reply by 20NE** [reply]
Totally agree OP, not just local router use case.
